test/geomap.py

```diff
@@ -0,0 +1,119 @@
+import math 
+import numpy as np
+import pandas as pd
+import utm
+
+class MAP():
+  """
+  MAP loads data from a csv file (region.csv) placed in the data folder
+
+  """
+  _data = {}
+  _fileName = './data/region.csv'
+  
+  def __init__(self):
+    # loading data
+    self._data = pd.read_csv( self._fileName )
+    print( 'Data successfully loaded!' )
+
+  def Data(self):
+    "Returns raw data"
+    return( self._data )
+
+  def CentrePoint(self):
+    "Returns the point at the centre of the square and its versor" 
+    
+    vX = self._data['upperLeftLat'] - self._data['lowerRightLat']
+    vY = self._data['lowerRightLon'] - self._data['upperLeftLon']
+    m = (vX ** 2 + vY ** 2) ** 0.5
+
+    self._data['centreLat'] = vX / 2 + self._data['upperLeftLat']
+    self._data['centreLon'] = vY / 2 + self._data['lowerRightLon']
+        
+    self._data['centreLatVersor'] = vX / m
+    self._data['centreLonVersor'] = vY / m
+  
+    return( self._data [['centreLon', 'centreLat']], self._data [['centreLonVersor', 'centreLatVersor']] )
+  
+  def FindPoint (self, lon, lat): 
+    """
+      Finds the closest location on the map relative
+      to the given point (lon, lat) in utm format
+    """
+    condition = (self._data['upperLeftLat'] >= lat) & ( lat > self._data['lowerRightLat']) & (self._data['upperLeftLon'] < lon) & ( lon <= self._data['lowerRightLon'])
+    return ( self._data[ condition ] )
+
+  # def Utm(self):
+  #   "Converts latitude and longitude geolocation to utm"
+  #   # It has to be reviewed: The data is in a different format (+ddddddd.ddddd). Check utm.from_latlon documentation
+    
+  #   self._data[ ['uLX', 'uLY', 'uLUtmZoneNumber', 'uLUtmZoneLetter'] ] = self._data.apply( 
+  #     lambda row: pd.Series( 
+  #       list( utm.from_latlon( row['upperLeftLat'], row['upperLeftLon'] ) )
+  #     ),
+  #     axis = 1 
+  #   )
+    
+  #   self._data[ ['lRX', 'lRY', 'lRUtmZoneNumber', 'lRUtmZoneLetter'] ] = self._data.apply( 
+  #     lambda row: pd.Series( 
+  #       list( utm.from_latlon( row['lowerRightLat'], row['lowerRightLon'] ) )
+  #     ),
+  #     axis = 1 
+  #   )
+
+  #   return (
+  #     self._data
+  #   )
+
+  def MdtGradient(self, data, step):

```

Comment:
Ditto. I think there is too much similar codes, it's not clear to me the division you're doing in your code. Looks like there is a directory for tests (not in the sense of software development), but I think you shouldn't include it in the codebase. Perhaps you can perform it only locally.

Reply:
Removed duplicated code.